src/CodeGenerator/InjectorGenerator.php

```diff
@@ -90,39 +97,46 @@ public function __construct(
         $this->logger = $logger ?? new NullLogger();
     }
 
-    /**
-     * Generate injector
-     *
-     * @param array $factories
-     */
-    private function generateInjector(array $factories)
+    private function buildFromTemplate(string $templateFile, string $outputFile, array $replacements) : void
     {
-        $listFile = new FileGenerator();
-        $listFile->setFilename($this->outputDirectory . '/factories.php')
-            ->setDocBlock(new DocBlockGenerator('AUTO GENERATED FACTORY LIST'))
-            ->setBody('return ' . var_export($factories, true) . ';');
+        $template = file_get_contents($templateFile);
+        $code = strtr($template, $replacements);
+        $file = new SplFileObject($outputFile, 'w');
 
-        $class = new ClassGenerator('GeneratedInjector', $this->namespace);
-        $classFile = new FileGenerator();
+        $file->fwrite($code);
+        $file->fflush();
+    }
 
-        $loadFactoryCode = '$this->factories = require __DIR__ . \'/factories.php\';';
-        $class->setExtendedClass('\\' . AbstractInjector::class)
-            ->addMethod('loadFactoryList', [], MethodGenerator::FLAG_PUBLIC, $loadFactoryCode);
+    private function generateInjector() : void
+    {
+        $file = $this->outputDirectory . '/GeneratedInjector.php';
+        $replacements = [
+            '%namespace%' => $this->namespace ? "namespace {$this->namespace};\n" : '',
+        ];
 
-        $classFile->setFilename($this->outputDirectory . '/GeneratedInjector.php')
-            ->setDocBlock(new DocBlockGenerator('AUTO GENERATED DEPENDENCY INJECTOR'))
-            ->setNamespace($class->getNamespaceName())
-            ->setClass($class);
+        $this->buildFromTemplate(self::INJECTOR_TEMPLATE, $file, $replacements);
```

Comment:
Same comment here - `$file` and `$replacements` are not re-used, so their assigments could instead be inlined here.

Reply:
This one hasn't been addressed yet.